tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -571,468 +495,442 @@ def tridiag_forward(
 
-    assert np.allclose(expected + 1.0, out1)
-    assert np.allclose(expected + 2.0, out2)
-    assert np.allclose(expected + 3.0, out3)
+    cases.verify_with_default_data(
+        cartesian_case,
+        testee,
+        ref=lambda: (expected + 1.0, (expected + 2.0, expected + 3.0)),
+        comparison=lambda ref, out: np.all(out[0].array() == ref[0])
+        and np.all(out[1][0].array() == ref[1][0])
+        and np.all(out[1][1].array() == ref[1][1]),
+    )
 
 
-@pytest.mark.parametrize("forward", [True, False])
-def test_scan_nested_tuple_input(fieldview_backend, forward):
+def test_scan_nested_tuple_input(cartesian_case):
     init = 1.0
-    inp1 = gtx.np_as_located_field(KDim)(np.ones((size,)))
-    inp2 = gtx.np_as_located_field(KDim)(np.arange(0.0, size, 1))
-    out = gtx.np_as_located_field(KDim)(np.zeros((size,)))
+    k_size = cartesian_case.default_sizes[KDim]
+    inp1 = gtx.np_as_located_field(KDim)(np.ones((k_size,)))
+    inp2 = gtx.np_as_located_field(KDim)(np.arange(0.0, k_size, 1))
+    out = gtx.np_as_located_field(KDim)(np.zeros((k_size,)))
 
-    prev_levels_iterator = lambda i: range(i + 1) if forward else range(size - 1, i - 1, -1)
+    prev_levels_iterator = lambda i: range(i + 1)
     expected = np.asarray(
         [
             reduce(lambda prev, i: prev + inp1[i] + inp2[i], prev_levels_iterator(i), init)
-            for i in range(size)
+            for i in range(k_size)
         ]
     )
 
-    @gtx.scan_operator(axis=KDim, forward=forward, init=init, backend=fieldview_backend)
+    @gtx.scan_operator(axis=KDim, forward=True, init=init)
     def simple_scan_operator(carry: float, a: tuple[float, float]) -> float:
         return carry + a[0] + a[1]
 
-    simple_scan_operator((inp1, inp2), out=out, offset_provider={})
-
-    assert np.allclose(expected, out)
+    cases.verify(cartesian_case, simple_scan_operator, (inp1, inp2), out=out, ref=expected)
 
 
-def test_docstring():
-    a_I_float = gtx.np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
-
+def test_docstring(cartesian_case):
     @gtx.field_operator
-    def fieldop_with_docstring(a: gtx.Field[[IDim], float64]) -> gtx.Field[[IDim], float64]:
+    def fieldop_with_docstring(a: cases.IField) -> cases.IField:
         """My docstring."""
         return a
 
     @gtx.program
-    def test_docstring(a: gtx.Field[[IDim], float64]):
+    def test_docstring(a: cases.IField):
         """My docstring."""
         fieldop_with_docstring(a, out=a)
 
-    test_docstring(a_I_float, offset_provider={})
-
+    a = cases.allocate(cartesian_case, test_docstring, "a")()
+    inout = cases.allocate(cartesian_case, fieldop_with_docstring, "a")()
 
-def test_domain(fieldview_backend):
-    inp = gtx.np_as_located_field(IDim, JDim)(np.ones((size, size), dtype=float64))
-    out = gtx.np_as_located_field(IDim, JDim)(2 * np.ones((size, size), dtype=float64))
+    cases.verify(cartesian_case, test_docstring, a, inout=inout, ref=a)
```

Comment:
No separate `inout` field needed, `a` is the in/out parameter
```suggestion
    cases.verify(cartesian_case, test_docstring, a, inout=a, ref=a)
```